docs/recipes/react.md

```diff
@@ -25,22 +25,21 @@ export const toggleMachine = Machine({
 Using [React hooks](https://reactjs.org/hooks) makes it easier to use state machines with function components. You can either use the official [`@xstate/react`](https://github.com/davidkpiano/xstate/tree/master/packages/xstate-react) package, a community solution like [`use-machine` by Carlos Galarza](https://github.com/carloslfu/use-machine/), or implement your own hook to interpret and use XState machines:
 
 ```js
-import { useState, useMemo, useEffect } from 'react';
+import { useState, useRef, useEffect } from 'react';
 import { interpret } from 'xstate';
 
 export function useMachine(machine) {
   // Keep track of the current machine state
   const [current, setCurrent] = useState(machine.initialState);
 
   // Define the service (only once!)
-  const service = useMemo(
-    () =>
-      interpret(machine).onTransition(state => {
-        // Update the current machine state when a transition occurs
-        if (state.changed) {
-          setCurrent(state);
-        }
-      }),
+  const service = useRef(
```

Comment:
@gaearon your thoughts would be much appreciated here. We are using `useRef()` to create a XState's service instance.

@esrch [left a comment](https://github.com/davidkpiano/xstate/pull/396#issuecomment-475811430) saying that starting the service directly with useRef will recreate the service on each render, which might reduce performance and might increase the memory usage. 

What do you say? It's worth optimizing it?